GCMessengerSDKSample/app/src/main/java/com/genesys/cloud/messenger/sample/MainActivity.kt

```diff
@@ -360,8 +376,87 @@ class MainActivity : AppCompatActivity(), ChatEventListener {
                 this, "Chat availability status returned ${it.isAvailable}",
             )
         }
+    }private fun enablePushNotifications(accountInfo: AccountInfo) {
+        Log.d(TAG, "enablePush()")
+        if (ContextCompat.checkSelfPermission(this, PERMISSION_POST_NOTIFICATIONS) == PackageManager.PERMISSION_GRANTED) {
+            postNotificationsGranted()
+        } else {
+            requestPermission(PERMISSION_POST_NOTIFICATIONS)
+        }
+    }
+
+    private fun postNotificationsGranted() {
+        val deviceToken = retrieveDeviceTokenForPush()
+        if (deviceToken != null) {
+            Log.d(TAG, "deviceToken read successfully: $deviceToken")
+            /* Temporary solution */
+            toast(this, "setPushToken() will be called here", Toast.LENGTH_LONG)
+            /* Temporary solution */
+            // TODO GMMS-8030 - Call Messenger SDK to register for Push Notifications
+        } else {
+            Log.d(TAG, "deviceToken not found")
+            Toast.makeText(this, R.string.enable_push_failed_message, Toast.LENGTH_LONG).show()
+        }
+    }
+
+    private fun postNotificationsDenied() {
+        Toast.makeText(this,R.string.enable_push_failed_message, Toast.LENGTH_LONG).show()
+    }
+
+    private fun retrieveDeviceTokenForPush(): String? {
+        return runBlocking {
+            var token :String? = null
+            IOScope().launch {
+                token = Tasks.await(FirebaseMessaging.getInstance().token)
+                Log.d(TAG, "deviceToken received: $token")
+            }.join()
+            token
+        }
+    }
+
+    private fun disablePushNotifications(accountInfo: AccountInfo) {
+        // TODO GMMS-8092 - Call Messenger SDK to unregister from Push Notifications
     }
 
+    //region - permissions
+
+    // android.Manifest.permission.POST_NOTIFICATIONS needs API 33
```

Comment:
You can use[ @RequiresApi](https://developer.android.com/reference/androidx/annotation/RequiresApi) annotation to indicate API level required.